tusclient/uploader.py

```diff
@@ -211,7 +223,35 @@ def upload_chunk(self):
         """
         Upload chunk of file.
         """
+        self._retried = 0
         self._do_request()
         self.offset = int(self.request.response_headers.get('upload-offset'))
         msg = '{} bytes uploaded ...'.format(self.offset)
         print(msg)
+
+    def _do_request(self):
+        # TODO: Maybe the request should not be re-created everytime.
+        #      The request handle could be left open until upload is done instead.
+        self.request = TusRequest(self)
+        try:
+            self.request.perform()
+            self.verify_upload()
+        except TusUploadFailed as error:
+            self.request.close()
+            self._retry_or_cry(error)
+        finally:
+            self.request.close()
+
+    def _retry_or_cry(self, error):
+        if self.retries > self._retried:
+            time.sleep(self.retry_delay)
```

Comment:
Yes I get what you are saying. But I imagine this is the natural pattern for coding in javascript. And to have something similar with python might not be as easy. We could with [asyncio](https://docs.python.org/3/library/asyncio.html), but that's only available in python 3 and leaves python 2 out. We could also do this with threading, but I imagine that might be farfetched and better left implemented by users who require the functionality.